Sources/Receipt/ListReceiptViewPresenter.swift

```diff
@@ -29,25 +29,49 @@ final class ListReceiptViewPresenter {
     private unowned let view: ListReceiptView
 
     private var offset = 0
-    private let limit = 20
+    private let userReceiptLimit = 20
+    private var prepaidCardToken: String?
+    private let prepaidCardReceiptLimit = 20
+    private var prepaidCardReceiptCreatedAfter = Calendar.current.date(byAdding: .year, value: -1, to: Date())
 
     private var isLoadInProgress = false
     private(set) var areAllReceiptsLoaded = true
     private(set) var sectionData = [(key: Date, value: [HyperwalletReceipt])]()
 
     /// Initialize ListReceiptPresenter
-    init(view: ListReceiptView) {
+    init(view: ListReceiptView, prepaidCardToken: String? = nil) {
```

Comment:
Maybe it is better to implement enum with associated values in this case. 
e.g.:
...

    enum ListReceiptType {
        case user
        case prepaidCard(String)
    }

...

This allows us to extend our code to support `Account Receipts` in the future: `case account(String, String)` for `program-token` and `account-token`.

add some `private var listReceiptType: ListReceiptType`

change initializers:
...

    init(view: ListReceiptView) {
        self.view = view
        listReceiptType = .user
    }

    init(view: ListReceiptView, prepaidCardToken: String) {
        self.view = view
        self.listReceiptType = .prepaidCard(prepaidCardToken)
    }
...

in future we can add initializer for account receipts:
...
```
init(view: ListReceiptView, programToken: String, accountToken: String)  {
    self.view = view
    self.listReceiptType = .account(programToken, accountToken)
}
```
...

Reply:
Hi @ymoskovko-epam , this is a great idea and @jsaini-hw had this solution as the first poc for this ticket. However,  it's hard to name the enum. In your case, I saw you named enum to be ListReceiptType, but receipt type already means the transaction type such as, payment, transfer to bank or fee, etc. It's hard to categorize the receipt sources in our case.  We will absolutely follow the way you and Jaspreet proposed if we can come up with a better name for user, prepaidCard and account receipts